stl/inc/cstring

```diff
@@ -39,6 +39,7 @@ using _CSTD strlen;
 using _CSTD strncat;
 using _CSTD strncmp;
 using _CSTD strncpy;
+using _CSTD strnlen;

```

Comment:
Speaking as a maintainer, I don't believe that we should merge this change (or the corresponding test change in `tests/tr1/tests/cstring/test.cpp`). Microsoft's `::strnlen` is a non-Standard extension, and while we can't get rid of it, we're trying to avoid adding non-Standard extensions going forward (particularly those that involve names that aren't reserved/`_Ugly` as reserved names indicate that something inherently non-Standard is happening).

For precedent/guidance, the Standard itself strongly suggests that `namespace std` should be kept "clean". WG21-N4892 16.4.2.3 [headers]/10 says:

> Annex K of the C standard describes a large number of functions, with associated types and macros, which "promote safer, more secure programming" than many of the traditional C library functions. The names of the functions have a suffix of `_s`; most of them provide the same service as the C library function with the unsuffixed name, but generally take an additional argument whose value is the size of the result array. If any C++ header is included, it is implementation-defined whether any of these names is declared in the global namespace. (None of them is declared in namespace `std`.)

And this applies to [C11](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf) K.3.7.4.4 "The `strnlen_s` function". If this Annex K function shouldn't be in `std`, plain `::strnlen` shouldn't either.

Additionally, even in the distant past when the STL added non-Standard machinery, we did so in the `stdext` namespace (e.g. `stdext::hash_map`, `stdext::checked_array_iterator`) in order to make it clear that `std` is truly Standard. The only major exceptions to this policy that I'm aware of are the TR1-era names `mersenne_twister` etc. in `<random>` which are historical relics (and which we want to remove, but binary compatibility makes it difficult).

Ultimately, this change wouldn't make it easier for users to write portable code, so I believe this falls under the project's [non-goals](https://github.com/microsoft/STL#non-goals). (I do appreciate the attempt to increase consistency, though!)